eds_scikit/io/arrow.py

```diff
@@ -0,0 +1,57 @@
+import os
+
+import pandas as pd
+import pyarrow as pa
+
+
+class arrowConnector:  # pragma: no cover
+    def __init__(self, sql, database_name: str):
+        self.database_name = database_name
+        self.db = (
+            sql(f"DESCRIBE DATABASE EXTENDED {database_name}")
+            .filter("database_description_item=='Location'")
+            .collect()[0]
+            .database_description_value
+        )
+
+    def get_pd_fragment(
+        self,
+        table_name: str,
+    ):
+
+        self.path_table = os.path.join(self.db, table_name)
+
+        # Import the parquet as ParquetDataset
+        parquet_ds = pa.parquet.ParquetDataset(
+            self.path_table, use_legacy_dataset=False
+        )
+
+        # Partitions of ds
+        fragments = iter(parquet_ds.pieces)
+
+        yield from fragments
+
+    def get_pd_table(
+        self,
+        table_name=None,
+        types_mapper=None,
+        integer_object_nulls=True,
+        date_as_object=False,
+        person_ids=None,
+    ):
+
+        filtered = []
+        for fragment in self.get_pd_fragment(table_name=table_name):
+            table = fragment.to_table()
+
+            # Import to pandas the fragment
+            table_pd = table.to_pandas(
+                types_mapper=types_mapper,
+                integer_object_nulls=integer_object_nulls,
+                date_as_object=date_as_object,
+            )
+            if (person_ids is not None) and ("person_id" in table_pd.columns):
+                table_pd = table_pd[table_pd.person_id.isin(person_ids)]
+            filtered.append(table_pd)
```

Comment:
I don't think this works with large tables that don't fit in RAM (e.g. `measurement` when it has more than 200m rows). I'm leaning towards writing tables in a batch fashion:
```python
pq.write_dataset(table, dataset_name)
```
as we saw IRL. What do you think?

Reply:
Since in the end, the goal is to load the tables as pandas DataFrames, they should fit in memory during writing anyway !
And at no point is the full table present in RAM, only the table filtered on the provided person_ids. But yes maybe we could check the size of the final filtered DF

Reply:
1. In my understanding, a user might not need/want to convert a table to pandas right away and stick to koalas for further agregation.
2. Users can provide `person_ids` of arbitrary size, and checking RAM usage feels quite hacky.
3. I think we agree on the action to take. What I suggest is to materialize each fragment as a pandas dataframe, filter this sub dataframe and write it, so that we never need to materialize the full table as a dataframe. This should work even with our ancient version of pyarrow.

Reply:
3. is actually what's happening [here](https://github.com/aphp/eds-scikit/blob/arrow_connector/eds_scikit/io/hive.py#L327) !

```python
        for fragment in fragments:
            table = fragment.to_table()
            # Import to pandas the fragment
            df = table.to_pandas(
                types_mapper=types_mapper,
                integer_object_nulls=integer_object_nulls,
                date_as_object=date_as_object,
            )
            if (person_ids is not None) and ("person_id" in df.columns):
                df = df[df.person_id.isin(person_ids)]
            filtered.append(df)

        return pd.concat(filtered)
```

For 1., I agree that at some point we might want to directly work on the parquet files as per our Ibis / DuckDb / etc discussion, but personaly I thought of this arrow connector only as a way to quickly save data locally. I'm afraid that exposing "too much" of this connector now might introduce some confusion with the *normal* way of connecting to the database (via `HiveData`directly). What do you think ?